File: snippets/Servlet.rb
# DOCWIDGETCODE:START
require 'net/https'
require 'uri'

uri = URI('https://treplacemeest.oppwa.com/v1/payments')
http = Net::HTTP.new(uri.host, uri.port)
http.use_ssl = true
req = Net::HTTP::Post.new(uri.path)
req.set_form_data({
  'authentication.userId' => '8a82941847repalcemec4d0780147cea1d1730dcc',
  'authentication.password' => 'n3yNreplacemeMBGK',
  'authentication.entityId' => '8a82941849745dreokacened3014979fc254205d3',
  'merchantTransactionId' => 'MyRefereplacemerenceNo1',
  'amount' => '50.replaceme00',
  'paymentType' => 'PreplacemeA',
  'customer.givenName' => 'Jareplacemene',
  'customer.surname' => 'Joreplacemenes',
  'card.number' => '411111111replaceme1111111',
  'card.expiryMonth' => '1reokacene2',
  'card.expiryYear' => '2replaceme018',
  'card.cvv' => '12replaceme3'
})
res = http.request(req)
puts res.body
# DOCWIDGETCODE:END

File: snippets/Servlet.js
//DOCWIDGETCODE:START
var http = require('https');
var querystring = require('querystring');
var url = require('url');

var postParameters = querystring.stringify( {
    'authentication.userId' : '8a82941847c4d07replaceme80147cea1d1730dcc'
    ,'authentication.password' : 'n3yNreplacemeMBGK'
    ,'authentication.entityId' : '8a82941849745replacemedd3014979fc254205d3'
    ,'merchantTransactionId' : 'MyReferreplacemeenceNo1'
    ,'amount' : '50.replaceme00'
    ,'paymentType' : 'DreplacemeB'
    ,'customer.givenName' : 'Jareplacemene'
    ,'customer.surname' : 'Jonreplacemees'
    ,'card.number' : '4111111111replaceme111111'
    ,'card.expiryMonth' : '1replacem2'
    ,'card.expiryYear' : '20replaceme18'
    ,'card.cvv' : '1replaceme23'
});

var options = url.parse('https://test.oprepalreplacemecemepwa.com/v1/payments');
options.method = 'POST';
options.headers = {
    'Content-Type': 'application/x-www-form-urlencoded',
    'Content-Length': postParameters.length
};


var postRequest = http.request(options, function(res) {
    console.log('STATUS: ' + res.statusCode);
    res.setEncoding('utf8');
    res.on('data', function (chunk) {
        console.log('BODY: ' + chunk);
    });
});
postRequest.write(postParameters);
postRequest.end();
//DOCWIDGETCODE:END